src/api/procedures/setCustodian.ts

```diff
@@ -55,40 +55,30 @@ export async function prepareSetCustodian(
     });
   }
 
-  const address = signerToString(targetAccount);
-  const authorizationRequests = await currentIdentity.authorizations.getSent();
+  const identityDid = signerToString(targetIdentity);
+  const rawIdentity = new Identity({ did: identityDid }, context);
 
-  const hasPendingAuth = !!authorizationRequests.data.find(authorizationRequest => {
-    const {
-      target,
-      data,
-      data: { type },
-    } = authorizationRequest;
-
-    if (type === AuthorizationType.PortfolioCustody) {
-      const authorizationData = data as { value: NumberedPortfolio | DefaultPortfolio };
-
-      return (
-        signerToString(target) === address &&
-        !authorizationRequest.isExpired() &&
-        type === AuthorizationType.PortfolioCustody &&
-        authorizationData.value.uuid === portfolio.uuid
-      );
-    }
+  const authorizationRequests = await rawIdentity.authorizations.getReceived({
+    type: AuthorizationType.PortfolioCustody,
+    includeExpired: false,
+  });
 
-    return false;
+  const hasPendingAuth = authorizationRequests.find(authorizationRequest => {
+    const { issuer, data } = authorizationRequest;
+    const authorizationData = data as { value: NumberedPortfolio | DefaultPortfolio };
+    return rawIdentity.did === issuer.did && authorizationData.value.uuid === portfolio.uuid;
```

Comment:
You need to compare the issuer against the current identity, not against the target